### ISSUE TITLE
jsx-a11y/media-has-caption cannot detect multiple tracks

### ISSUE THREAD

**Issue by davidgoli** [jsx-eslint/eslint-plugin-jsx-a11y#838]
The following code fails the rule `media-has-caption` with the default configuration:

```ts
      <video
        {...props}
        ref={videoRef}
        muted={autoMuted}
        data-testid={`video-${videoId}`}
        poster={poster}
        onEnded={onEnded}
      >
        {tracks.map((trackData) => (
          <track key={trackData.src} {...trackData} />
        ))}
      </video>
```

(where `tracks` is an array of track data)

Expected behavior: 
When media data is dynamically loaded & rendered, `<track>` tags should be recognized. In this case, there is of course no guarantee at compile time that the data will actually contain a track - but that's an issue with the data source, not the markup.

**Comment by ljharb** [comment]
What is the exact warning you're seeing, and on which line does it indicate?

**Comment by davidgoli** [comment]
The warning is:

```
  72:7  error  Media elements such as <audio> and <video> must have a <track> for captions  jsx-a11y/media-has-caption
```

It indicates the line on which the `<video>` tag occurs

**Comment by ljharb** [comment]
gotcha, thanks.

i think that unless we can know for certain that the array is non-empty, it'd be better to let this be a "use an override comment" situation.

However, with TS types, we may be able to ensure that.

**Comment by backwardok** [comment]
For the purposes of validating that tracks are being used in a video element, I think supporting dynamic track declarations like this should be allowed. Given that the rule doesn't actually validate that there's an actual captions file amongst the tracks, just that a track element exists, I think this is a meaningful extension. ESLint override comments seem like they should be more used for rare cases and/or exceptions, and this feels like a common case where you'd potentially have multiple subtitle tracks and a captions track.

**Comment by ljharb** [comment]
It'd certainly be fine to add an option that allows this, so that those who need it can opt into it explicitly.
